### ISSUE TITLE
`manual_is_power_of_two` is too broad for a default lint

### ISSUE THREAD

**Issue by newpavlov** [rust-lang/rust-clippy#13547]
### Description

In my project I have a bunch of code like this:
```rust
use static_assertions as sa;

const MY_FLAG: u32 = const {
    // const computation
};

// Enforce that the flag mask contains only one bit
sa::const_assert_eq!(MY_FLAG.count_ones(), 1);
```

The new `manual_is_power_of_two` lint gets triggered by the assert. Changing it to `is_power_of_two()` as suggested by the lint would make the intent less clear.

### Version

rustc 1.84.0-nightly (27861c429 2024-10-13)
binary: rustc
commit-hash: 27861c429af736ea3a6bb015956c7286071b286d
commit-date: 2024-10-13
host: x86_64-unknown-linux-gnu
release: 1.84.0-nightly
LLVM version: 19.1.1

**Comment by daira** [comment]
The stated rationale applies for `x.count_ones() == 1` but not for `x & (x - 1) == 0`. But note that the latter is not equivalent to `x.count_ones() == 1` or `x.is_power_of_two()`: it gives a different result for `x == 0`, and the lint message doesn't point that out. So I agree with the outcome of moving it to `pedantic`.

Even better would be to split it into two lints that trigger separately on `x.count_ones() == 1` and `x & (x - 1) == 0`. The former can be pedantic. The latter should have a message that says that the behaviour for `x == 0` *might* be unintended, as well as pointing out the lack of clarity.
